src/common/linkedlist.c

```diff
@@ -75,23 +75,22 @@ void linked_list_append_func(linked_list_s_t* list, generic_fn_t func) {
 	it->next = n;
 }
 
-// NOTE: this will likely break if the list has intermixed data and function ptrs
-// TODO: add tag field to ll_node_s_t to easily check what is present
 void linked_list_remove_func(linked_list_s_t* list, generic_fn_t func) {
 	if (list == NULL || list->head == NULL) return;
 
 	ll_node_s_t* it = list->head;
 	ll_node_s_t* p = NULL;
 	while (it != NULL) {
-		if (it->payload.func == func) {
-			if (p == NULL)
-				list->head = it->next;
-			else
-				p->next = it->next;
-			kfree(it);
+		if(it->isFunc) { //Outer if loop for protective wrapping
+			if (it->payload.func == func) {
```

Comment:
Shouldn't the pointer be enough to see what path we should take? Eg the caller knows based on context.

Reply:
It's just a pointer compare, so I don't see how it would matter anyway.

Reply:
Makes sense, I guess we could leave this here in the case that we may need to create a mixed function/data LL somewhere down the line. It'll just be a future proofing change that may be merged if necessary.

Reply:
Just to be clear, the the potential error here is that if there is a non-pointer added to the union, then it could potentially be re-interpreted casted to a valid pointer and be removed accidentally.

Perhaps just updating the comments would be good enough, and putting one on the union in ll_node_s on what's going on here.